build/android/generate_android_project.gni

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/android/rules.gni")
+
+# Runs generate_xwalk_core_library.py.
+#
+# Args:
+#   binary_files [optional]
+#     List of additional binary data (icudtl.dat, for example) that will be
+#     added to res/raw.
+#   build_config [required]
+#     Path to a .build_config file with resource and native libraries
+#     information (generally, a build_config with the "android_apk" type).
+#   has_native_libraries [optional]
+#     Boolean indicating whether there are architecture-dependent shared
+#     libraries to copy to the project.
+#   js_bindings [required]
+#     List of Crosswalk extension bindings .js files to add to res/raw.
+#   output_dir [required]
+#     The project directory to be created. If it already exists, it will be
+#     erased.
+#   template_dir [required]
+#     Directory to the template directory containing a ant.properties,
+#     AndroidManifest.xml and others.
+template("generate_xwalk_core_library") {
+  _build_config = invoker.build_config
```

Comment:
For all "required" parameters, how about to add assert like `assert(defined(invoker.build_config))`?  just like `src/build/config/android/rules.gni` does.